lib/Doctrine/DBAL/Schema/AbstractAsset.php

```diff
@@ -180,7 +181,7 @@ public function getQuotedName(AbstractPlatform $platform)
         $keywords = $platform->getReservedKeywordsList();
         $parts    = explode('.', $this->getName());
         foreach ($parts as $k => $v) {
-            $parts[$k] = $this->_quoted || $keywords->isKeyword($v) ? $platform->quoteIdentifier($v) : $v;
+            $parts[$k] = $this->_quoted || $keywords->isKeyword($v) || preg_match('/\W/', $v) ? $platform->quoteIdentifier($v) : $v;
```

Comment:
This is a slippery slope. An object name needs to be quoted if otherwise it would be interpreted as an SQL expression which is far beyond `preg_match('/\W/', $v)`. Additionally, on some of the supported platforms (e.g. Oracle) quoted object names have different semantics than unquoted ones.

In addition to that, given https://github.com/doctrine/dbal/issues/3152 and https://github.com/doctrine/orm/issues/7815, DBAL is still a leaky abstraction when it comes to handling arbitrary object names.

Before we proceed with any changes like this, I'd like to explicitly define what we do and what we do not expect from DBAL regarding quoting object names.

After that, we’ll need _functional cross-platform_ tests that prove that this promise is fulfilled. Testing SQL without executing it is not enough.

Reply:
@morozovm, my argument here is that object names with non-word characters are going to blow up anyhow, so DBAL might as well quote them, regardless of any potential semantic changes on a given platform.  Quoting an object name and ending up with a case-sensitive field name cannot possibly be worse than the syntax error that occurs running the SQL if you don't quote it.  If it is OK to quote for keywords, then it should be absolutely fine to also do it for non-word characters.

My use case is using the Laravel Schema Builder to dynamically import multiple Access DBs into MySQL.  Laravel does something funny with the quoting, so quoting the field names myself at that level was not enough to work around this.  `preg_match('/\W/', $v)` was more than enough to deal with that use case, and I don't see how functional tests could make this particular patch any better.  Every single field name that this quotes was enough to generate invalid SQL before the patch.  Can you provide me with an example of a SQL expression that _does not_ contain a non-word character?  How about an example of a valid column name that _does_?

Thanks.

Reply:
I agree that there won't be a BC break if we start quoting some names that otherwise would have been interpreted as expressions and fail. My concern is that there are no clear expectations about how DBAL should handle such names and where. And it's not tested in an integration way (i.e. unit tests may claim that it works but it practice it doesn't because such SQL never gets run against a DB).

`preg_match('/\W/', $v)` will work for the names containing whitespaces but it's just a particular case. E.g. the following is still not handled:

```sql
CREATE TABLE `1` (
    `2` INTEGER
)
```

There must be added functional tests handling "special" names around all schema introspection/deployment use cases before we consider it supported.

Reply:
This doesn't seem like a good reason to hold up this PR to me.  This PR adds additional functionality without breaking anything, fixing SQL that was previously broken.

Reply:
Lack of proper tests is a valid reason to not accept code changes. Please stop arguing and collaborate.

Reply:
I've collaborated for weeks on this PR.  In fact, I think I've been very accommodating, rewriting and expanding this patch for @greg0ire and @SenseException several times.  I agree that integration tests would be more thorough, but you agree that this minor change should not have broken anything and it is being tested as thoroughly as that area of the code was tested previous to my relatively minor change.  Please merge this and do not hold this useful contribution hostage to my making additional useful contributions immediately.

Reply:
this should be more like `! preg_match('~^\w+$~', ...)`